pkg/handler/collector/oci/oci.go

```diff
@@ -183,89 +208,192 @@ func (o *ociCollector) getTagsAndFetch(ctx context.Context, repo string, tags []
 // A workaround for this would be to run the collector again with a specific tag without polling and ingest like normal
 func (o *ociCollector) fetchOCIArtifacts(ctx context.Context, repo string, rc *regclient.RegClient, image ref.Ref, docChannel chan<- *processor.Document) error {
 	logger := logging.FromContext(ctx)
-
 	// attempt to request only the headers, avoids Docker Hub rate limits
 	m, err := rc.ManifestHead(ctx, image)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed retrieving manifest head: %w", err)
 	}
 
 	if m.IsList() {
 		m, err := rc.ManifestGet(ctx, image)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed retrieving manifest: %w", err)
 		}
+
 		pl, _ := manifest.GetPlatformList(m)
+		logger.Infof("%s is manifest list with %d platforms", image.Reference, len(pl))
+
+		// Use goroutines to fetch platforms concurrently
+		// Create a channel to collect errors from goroutines
+		errorChan := make(chan error, len(pl))
```

Comment:
While reviewing, I noticed a subtle concurrency issue that might lead to a deadlock in specific scenarios. 

Specifically, in the concurrent error handling section, there are two lines within the goroutine where an error might be sent to `errorChan`. 

However, errorChan is buffered to only len(pl), potentially causing the goroutine to block if two errors are sent.

```go
if err != nil {
	errorChan <- fmt.Errorf("failed retrieving platform specific digest: %w", err)
}
// ...
if err := o.fetchOCIArtifacts(ctx, repo, rc, platformImage, docChannel); err != nil {
	errorChan <- fmt.Errorf("failed retrieving platform specific digest: %w", err)
}
```

## Solution
``` go
if err != nil {
 errorChan <- fmt.Errorf("failed retrieving platform specific digest: %w", err)
 return // exit early
}
```

Reply:
great catch, will fix